### PR TITLE
EMP effect for Silicons

### PR DESCRIPTION
**Pull request by Pspritechologist**
# Description
General fixes and additions to the Silicon stuff.

# TODO
- [x] Add EMP effects to Silicons
- [x] Add solar flare effects to Silicons
- [ ] Markings?

# Changelog

:cl:
- add: IPCs (and other Silicons) do not appreciate EMP grenades :)
- fix: IPCs can now be healed of bleeding, the same way you heal anything else. A way to put the coolant back in them soon...

### CODE REVIEW

File: Content.Server/StationEvents/Components/SolarFlareRuleComponent.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/StationEvents/Components/SolarFlareRuleComponent.cs:1]:
```suggestion
using System.Numerics; // Parkstation-Ipc
```

File: Content.Server/SimpleStation14/Silicon/Systems/SiliconMiscSystem.cs
Comment (DEATHB4DEFEAT) [inline Content.Server/SimpleStation14/Silicon/Systems/SiliconMiscSystem.cs:7]:
```suggestion
namespace Content.Server.SimpleStation14.Silicon.Systems;
```

Comment (Pspritechologist) [reply Content.Server/SimpleStation14/Silicon/Systems/SiliconMiscSystem.cs:7]: This would require a more major refactoring of the Silicon files and name spaces. I'll do it at some point, but it probably shouldn't hold up this pr.

Comment (DEATHB4DEFEAT) [reply Content.Server/SimpleStation14/Silicon/Systems/SiliconMiscSystem.cs:7]: what

File: Content.Server/Emp/EmpSystem.cs
Comment (Pspritechologist) [inline Content.Server/Emp/EmpSystem.cs]:
```suggestion
            var ev = new EmpPulseEvent(energyConsumption, false, false, TimeSpan.FromSeconds(duration)); // Parkstation-IPCs
```

File: Content.Server/Emp/EmpSystem.cs
Comment (Pspritechologist) [inline Content.Server/Emp/EmpSystem.cs]:
```suggestion
public record struct EmpPulseEvent(float EnergyConsumption, bool Affected, bool Disabled, TimeSpan Duration); // Parkstation-IPCs
```

**Comment by Pspritechologist** [comment]
I think this is ready to merge? Sanity check it when it's reviewed.

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Pull request merged**
